core/jobs/batch_jobs/suggestion_stats_computation_jobs.py

```diff
@@ -97,77 +149,255 @@ def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+            | 'Group skill opportunity by ID' >> beam.GroupBy(lambda m: m.id)
+        )
 
-        user_stats_results = (
+        exp_opportunity_to_submitted_suggestions = (
             {
-                'suggestion': suggestions_grouped_by_target,
+                'suggestion': translation_suggestions_grouped_by_target,
                 'opportunity': exp_opportunities
             }
             | 'Merge models' >> beam.CoGroupByKey()
-            | 'Get rid of key' >> beam.Values()  # pylint: disable=no-value-for-parameter
-            | 'Generate stats' >> beam.ParDo(
-                lambda x: self._generate_stats(
+            | 'Get rid of key submitted objects' >> beam.Values()  # pylint: disable=no-value-for-parameter
+        )
+        exp_opportunity_to_reviewed_submitted_suggestions = (
+            {
+                'suggestion': (
+                    reviewed_translation_suggestions_grouped_by_target),
+                'opportunity': exp_opportunities
+            }
+            | 'Merge reviewed models' >> beam.CoGroupByKey()
+            | 'Get rid of key of reviewed objects' >> beam.Values()  # pylint: disable=no-value-for-parameter
+        )
+        skill_opportunity_to_submitted_suggestions = (
+            {
+                'suggestion': question_suggestions_grouped_by_target,
+                'opportunity': skill_opportunities
+            }
+            | 'Merge submitted question models' >> beam.CoGroupByKey()
+            | 'Get rid of key of submitted question objects' >> beam.Values()  # pylint: disable=no-value-for-parameter
+        )
+        skill_opportunity_to_reviewed_suggestions = (
+            {
+                'suggestion': question_reviews_grouped_by_target,
+                'opportunity': skill_opportunities
+            }
+            | 'Merge reviewed question models' >> beam.CoGroupByKey()
+            | 'Get rid of key of reviewed question objects' >> beam.Values()  # pylint: disable=no-value-for-parameter
+        )
+
+        translation_contribution_stats_keys_and_results = (
+            exp_opportunity_to_submitted_suggestions
+            | 'Generate translation contribution stats' >> beam.ParDo(
+                lambda x: self._generate_translation_stats(
                     x['suggestion'][0] if len(x['suggestion']) else [],
                     list(x['opportunity'][0])[0]
-                    if len(x['opportunity']) else None
+                    if len(x['opportunity']) else None,
+                    suggestion_models.TranslationContributionStatsModel
+                ))
+        )
+        translation_review_stats_keys_and_results = (
+            exp_opportunity_to_reviewed_submitted_suggestions
+            | 'Generate translation review stats' >> beam.ParDo(
+                lambda x: self._generate_translation_stats(
+                    x['suggestion'][0] if len(x['suggestion']) else [],
+                    list(x['opportunity'][0])[0]
```

Comment:
Isn't `x['opportunity']` a `list`? Why can't we just do `x['opportunity'][0]`? Ditto elsewhere.